commander3/src/comm_tod_WMAP_mod.f90

```diff
@@ -19,469 +19,1237 @@
+   integer(i4b), parameter :: samp_rcal = 19
+   integer(i4b), parameter :: prep_relbp = 5
+   integer(i4b), parameter :: prep_absbp = 16
+   integer(i4b), parameter :: samp_bp = 11
+   integer(i4b), parameter :: samp_sl = 6
+   integer(i4b), parameter :: samp_N_par = 7
+   integer(i4b), parameter :: sel_data = 8
+   integer(i4b), parameter :: bin_map = 9
+   integer(i4b), parameter :: calc_chisq = 10
+   integer(i4b), parameter :: output_slist = 12
+   integer(i4b), parameter :: samp_mono = 13
+   integer(i4b), parameter :: sub_sl      = 14
+   integer(i4b), parameter :: sub_zodi = 17
+   integer(i4b), parameter :: sim_map = 20
+   integer(i4b), parameter :: samp_imbal = 21
+   integer(i4b), parameter :: samp_bline = 22
+   logical(lgt), dimension(N_test) :: do_oper
+
+   type, extends(comm_tod) :: comm_WMAP_tod
+      class(orbdipole_pointer), allocatable :: orb_dp ! orbital dipole calculator
+      character(len=20), allocatable, dimension(:) :: labels ! names of fields
    contains
-     procedure     :: process_tod        => process_WMAP_tod
-  end type comm_WMAP_tod
+      procedure     :: process_tod => process_WMAP_tod
+   end type comm_WMAP_tod
 
-  interface comm_WMAP_tod
-     procedure constructor
-  end interface comm_WMAP_tod
+   interface comm_WMAP_tod
+      procedure constructor
+   end interface comm_WMAP_tod
 
 contains
 
-  !**************************************************
-  !             Constructor
-  !**************************************************
-  function constructor(cpar, id_abs, info, tod_type)
-    implicit none
-    type(comm_params),       intent(in) :: cpar
-    integer(i4b),            intent(in) :: id_abs
-    class(comm_mapinfo),     target     :: info
-    character(len=128),      intent(in) :: tod_type
-    class(comm_WMAP_tod),    pointer    :: constructor
-
-    integer(i4b) :: i,nside_beam, lmax_beam, nmaps_beam, ndelta
-    character(len=512) :: datadir
-    logical(lgt) :: pol_beam
-
-    ! Set up WMAP specific parameters
-    allocate(constructor)
-    constructor%output_n_maps = 3
-    constructor%samprate_lowres = 1.d0  ! Lowres samprate in Hz
-    constructor%nhorn    = 1
-
-    ! Initialize beams
-    nside_beam = 512
-    nmaps_beam = 3
-    pol_beam   = .true.
-    constructor%nside_beam = nside_beam
-
-
-    !initialize the common tod stuff
-    call constructor%tod_constructor(cpar, id_abs, info, tod_type)
-    
-    !TODO: this is LFI specific, write something here for wmap
-    call get_tokens(cpar%ds_tod_dets(id_abs), ",", constructor%label)
-    do i = 1, constructor%ndet
-       if (mod(i,2) == 1) then
-          constructor%partner(i) = i+1
-       else
-          constructor%partner(i) = i-1
-       end if
-       constructor%horn_id(i) = (i+1)/2
-    end do
+   !*************************************************
+   !    Convert integer to string
+   !*************************************************
+   character(len=20) function str(k)
```

Comment:
This should be moved to comm_utils

Reply:
it's possible there's something like this already in the codebase already, but can defo move it there